### PR TITLE
Disable skip financial aid button during API activity

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Part of #1752
#### What's this PR do?
- Disable skip financial aid button during API activity
- Disable skip or add financial aid buttons when the other one has API activity going on, without showing a spinner

### CODE REVIEW

**Comment by aliceriot** [comment]
functionality looks great!

File: static/js/components/SkipFinancialAidDialog.js [hunk 1]
```diff
@@ -3,39 +3,47 @@ import React from 'react';
...
+      component={Button}
+      spinning={fetchSkipStatus === FETCH_PROCESSING}
+      disabled={fetchAddStatus === FETCH_PROCESSING}
       type='button'
-      className="primary-button save-button"
+      className="primary-button save-button skip-button"
       onClick={skip}>
       Pay Full Price
-    </Button>
+    </SpinnerButton>
```
Comment (aliceriot) [inline static/js/components/SkipFinancialAidDialog.js]: could we replace these two elements with a call to `dialogActions`?

File: static/js/containers/FinancialAidCalculator.js [hunk 2]
```diff
@@ -166,7 +173,11 @@ const FinancialAidCalculator = ({
     autoScrollBodyContent={true}
     onRequestClose={closeDialogAndCancel}
     actions={calculatorActions(
-      openConfirmSkipDialog, closeDialogAndCancel, () => saveFinancialAid(financialAid), fetchAddStatus
+      openConfirmSkipDialog,
+       closeDialogAndCancel,
+        () => saveFinancialAid(financialAid),
+        fetchAddStatus,
+        fetchSkipStatus,
```
Comment (aliceriot) [inline static/js/containers/FinancialAidCalculator.js]: can you fix the indentation here?

File: static/js/components/SkipFinancialAidDialog.js (hunk 1 above)
Comment (noisecapella) [inline static/js/components/SkipFinancialAidDialog.js]: The `<div>` with the `actions` class was preventing me, but I can take a look and see if I can work around that

Comment (noisecapella) [inline static/js/components/SkipFinancialAidDialog.js]: It looks like the `actions` class wasn't doing anything important in this case so I just swapped it with `dialogActions`

Comment (aliceriot) [inline static/js/components/SkipFinancialAidDialog.js]: I think that `div` was just there to have a wrapping element, not for any actual functionality.

File: static/js/components/inputs/util.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ import SpinnerButton from '../SpinnerButton';
  * Helper function to create dialog action buttons, with a SpinnerButton for the save button
  */
 export const dialogActions = (
-  onCancel: Function, onSave: Function, inFlight: bool, text: string='Save', saveClass: string=''
+  onCancel: Function, onSave: Function, inFlight: bool, text: string='Save', saveClass: string='', disabled: bool=false
```
Comment (aliceriot) [inline static/js/components/inputs/util.js]: is the annotation for `text`, `saveClass` and `disabled` still necessary when they're provided with a default value? I feel like Flow should be able to infer their types from that

Comment (noisecapella) [inline static/js/components/inputs/util.js]: No, just wanted to be explicit

**Pull request merged**